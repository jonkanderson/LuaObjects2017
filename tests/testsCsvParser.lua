local args = {...}
local CL, FN, COL, S = dofile('../root.lua')

--[[
Todo:
- Create real assertions.
--]]

local Pm = FN.newPathManager()
Pm:setArgs(args)

Pm:setModDir('../mods')
Pm:loadMod('modScanner.lua')
Pm:loadMod('modCsvParser.lua')

--====================================================== main code
-- Good for debugging.
--FN.reportAllKeys()

local dataDir = "./testsCsvParser.data"
local pr = CL.CsvParser:newInstance()
local tokens

function myString(obj)
	if type(obj) == "string" then return obj
	--elseif obj:isCsvHeader() then return obj:getKeyString()
	elseif obj:isCsvHeader() then return obj:debugString()
	else return obj:asCsvString() end
	--else return obj:debugString() end
end

function printOrderedTable(t)
	print("--[[ printOrderedTable")
	for _,ea in ipairs(t) do print(myString(ea)) end
	print("--]]")
end

function printKeyedTable(t)
	print("--[[ printKeyedTable")
	for k,ea in pairs(t) do print(myString(k).." -> "..myString(ea)) end
	print("--]]")
end

--====================================================== libreoffice01.csv
local fn = dataDir.."/libreoffice01.csv"
pr:openFile(fn)
print("Testing data: "..fn)

local header = pr:readBasicHeader()
printOrderedTable(header)

local t
t = pr:readReadRowOrdered()
printOrderedTable(t)

t = pr:readReadRowKeyed()
printKeyedTable(t)

while t do
	t = pr:readReadRowOrdered()
	if t then printOrderedTable(t) end
end

pr:close()

--====================================================== bb01.csv
-- This file has a Unicode BOM.
local fn = dataDir.."/bb01.csv"
pr:openFile(fn)
print("Testing data: "..fn)

local header = pr:readBbStyleHeader()
printOrderedTable(header)

t = true
while t do
	t = pr:readReadRowKeyed()
	if t then printKeyedTable(t) end
end

pr:close()
