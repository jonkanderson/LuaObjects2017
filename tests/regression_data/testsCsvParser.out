



--]]
--]]
--]]
--]]
--]]
--]]
--]]
--]]
--]]
--]]
--]]
_01
_01 ->     
_02
_02 -> 
1
10
12
  23  
2.34
=23+45
2.345
3
37038239146735200000000000000000000
a
A081: Confidence Interval 1(72927,Total Pts: 12)
A081: Confidence Interval 1(72927,Total Pts: 12) -> 12.00
A081: Confidence Interval 1(72927,Total Pts: 12) -> 7.00
“aaa”
Analysis Paper 1(71242,Total Pts: 80)
Analysis Paper 1(71242,Total Pts: 80) -> 
Analysis Paper 1(71242,Total Pts: 80) -> Needs Grading
Availability
Availability -> Yes
Availability -> Yes
CategoryA
CategoryA -> aa
Category B
Category B -> 2
Category C
Category C -> 1.4142135624
Category_D [x=2, y='cool']
Category_D [x=2, y='cool'] -> I have a quote: "
Extra column
First Name
First Name -> John
First Name -> Matt
Identifier
Identifier -> 11
Last Access
Last Access -> 2016-02-20 18:52:54
Last Access -> 2016-02-24 11:35:26
Last Name
Last Name -> Cool
Last Name -> Doe
   "Manual modified fields with spaces: LibreOffice parser error adds these space incorrectly:"    
--[[ printKeyedTable
--[[ printKeyedTable
--[[ printKeyedTable
--[[ printOrderedTable
--[[ printOrderedTable
--[[ printOrderedTable
--[[ printOrderedTable
--[[ printOrderedTable
--[[ printOrderedTable
--[[ printOrderedTable
--[[ printOrderedTable
Student ID
Student ID -> 
Student ID -> 
=SUM(C2:C4)
Testing data: ./testsCsvParser.data/bb01.csv
Testing data: ./testsCsvParser.data/libreoffice01.csv
Username
Username -> john.doe
Username -> matt.cool
Weighted Total(66663,Total Pts: up to 367)
Weighted Total(66663,Total Pts: up to 367) -> 56.83%
Weighted Total(66663,Total Pts: up to 367) -> 93.46%
xx
